File: artifacts/api-server/src/routes/feedback.ts
import { Router } from "express";
import { db, feedbackTable, usersTable } from "@workspace/db";
import { requireAuth, requireRole } from "../middlewares/auth";
import { SubmitFeedbackBody } from "@workspace/api-zod";

const router = Router();

router.post("/feedback", requireAuth, requireRole("student"), async (req, res): Promise<void> => {
  const parsed = SubmitFeedbackBody.safeParse(req.body);
  if (!parsed.success) {
    res.status(400).json({ error: parsed.error.message });
    return;
  }

  const [fb] = await db.insert(feedbackTable).values({
    eventId: parsed.data.eventId,
    studentId: req.user!.userId,
    rating: parsed.data.rating,
    comments: parsed.data.comments ?? null,
    customResponses: parsed.data.customResponses ?? null,
  }).returning();

  const [student] = await db.select().from(usersTable).where(
    (await import("drizzle-orm")).eq(usersTable.id, req.user!.userId)
  );

  res.status(201).json({
    id: fb.id,
    eventId: fb.eventId,
    studentId: fb.studentId,
    rating: fb.rating,
    comments: fb.comments ?? null,
    customResponses: fb.customResponses ?? null,
    submittedAt: fb.submittedAt.toISOString(),
    student: student ? {
      id: student.id,
      name: student.name,
      email: student.email,
      role: student.role,
      department: student.department,
      createdAt: student.createdAt.toISOString(),
    } : null,
  });
});

export default router;

File: artifacts/api-server/routes/feedback.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from pydantic import BaseModel
from typing import Optional
from database import get_db
import models
from auth_utils import require_roles

router = APIRouter()

class FeedbackBody(BaseModel):
    eventId: int
    rating: int
    comments: Optional[str] = None
    customResponses: Optional[str] = None

@router.post("/feedback", status_code=201)
def submit_feedback(
    body: FeedbackBody,
    current_user: models.User = Depends(require_roles("student")),
    db: Session = Depends(get_db),
):
    fb = models.Feedback(
        event_id=body.eventId,
        student_id=current_user.id,
        rating=body.rating,
        comments=body.comments,
        custom_responses=body.customResponses,
    )
    db.add(fb)
    db.commit()
    db.refresh(fb)
    return {
        "id": fb.id, "eventId": fb.event_id, "studentId": fb.student_id,
        "rating": fb.rating, "comments": fb.comments,
        "customResponses": fb.custom_responses,
        "submittedAt": fb.submitted_at.isoformat() if fb.submitted_at else "",
        "student": {
            "id": current_user.id, "name": current_user.name,
            "email": current_user.email, "role": current_user.role,
            "department": current_user.department,
            "createdAt": current_user.created_at.isoformat() if current_user.created_at else "",
        },
    }
